javascript/index.js

```diff
@@ -23,18 +23,138 @@
   getInstruction("mashedPotatoes", 4, (step5) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
     document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
+  }, (error) => console.log(error)); */
 
 
 
 // Iteration 1 - using callbacks
-// ...
+
+setTimeout(() => {
+  getInstruction("mashedPotatoes", 0, (step0) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step0}</li>`;
+  });
+}, 0);
+
+setTimeout(() => {
+  getInstruction("mashedPotatoes", 1, (step1) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+  });
+}, 1001);
+
+setTimeout(() => {
+  getInstruction("mashedPotatoes", 2, (step2) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+  });
+}, 2002);
+
+setTimeout(() => {
+  getInstruction("mashedPotatoes", 3, (step3) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+  });
+}, 3003);
+
+setTimeout(() => {
+  getInstruction("mashedPotatoes", 4, (step4) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+    document.querySelector("#mashedPotatoes").innerHTML +=
+      "<li>Mashed potatoes are ready!</li>";
+    document.getElementById("mashedPotatoesImg").removeAttribute("hidden");
+  });
+}, 4004);
 
 // Iteration 2 - using promises
-// ...
+
+obtainInstruction("steak", 0)
```

Comment:
this could also be recursive. 

<details>
<summary>potential solution: </summary>

```js
function cookRecipe(recipe, step = 0) {
  obtainInstruction(recipe, step)
    .then((result) => {
      document.querySelector("#" + recipe).innerHTML += `<li>${result}</li>`;
      cookRecipe(recipe, step + 1);
    })
    .catch((err) => console.log(err));
}

cookRecipe("steak");
```

</details>